pkg/controller/config.go

```diff
@@ -0,0 +1,147 @@
+package controller
```

Comment:
feels weird to me we have a private internal config and another one here. Can you elaborate on the reasons for this?